File: app/javascript/controllers/session_controller.js
import { Controller } from "@hotwired/stimulus"

export default class extends Controller {
  static values = { url: String }

  login_cliente() {
    this._send_post_request(this.urlValue)
  }

  login_vendedor() {
    this._send_post_request(this.urlValue)
  }

  login_admin() {
    this._send_post_request(this.urlValue)
  }

  logout() {
    this._send_post_request(this.urlValue)
  }

  _send_post_request(url) {
    fetch(url, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        "X-CSRF-Token": document.querySelector('[name="csrf-token"]').content,
      },
    })
      .then((response) => response.json())
      .then((data) => {
        if (data.session) {
          sessionStorage.setItem("session", JSON.stringify(data.session))
          window.location.href = "/"
        } else {
          sessionStorage.removeItem("session")
          window.location.href = "/login"
        }
      })
      .catch((error) => {
        console.error(error)
      })
  }
}

File: app/controllers/session_controller.rb
class SessionController < ApplicationController
  def login_cliente
    usuario = Usuario.find_by!(cargo: "cliente")
    session[:usuario_id] = usuario.id
    render json: {}
  end

  def login_vendedor
    usuario = Usuario.find_by!(cargo: "vendedor")
    session[:usuario_id] = usuario.id
    render json: {}
  end

  def login_admin
    usuario = Usuario.find_by!(cargo: "admin")
    session[:usuario_id] = usuario.id
    render json: {}
  end

  def logout
    session[:usuario_id] = nil
    render json: {}
  end
end
